Gems/Atom/Feature/Common/Code/Source/Mesh/StreamBufferViewsBuilder.cpp

```diff
@@ -188,21 +281,40 @@ namespace AZ
         }
 
         AZ::RHI::Ptr<AZ::RHI::BufferView> ShaderStreamBufferViewsBuilder::BuildShaderIndexBufferView(
-            const AZ::RPI::ModelLod::Mesh& modelLodMesh) const
+            const AZ::RPI::ModelLod::Mesh& modelLodMesh, AZ::RHI::IndexBufferView& indexBufferViewOut) const
         {
             AZ_Assert(modelLodMesh.GetDrawArguments().m_type == AZ::RHI::DrawType::Indexed, "We only support indexed geometry!");
             const AZ::RHI::IndexBufferView& indexBufferView = modelLodMesh.GetIndexBufferView();
+            indexBufferViewOut = indexBufferView;
 
             uint32_t indexElementSize = indexBufferView.GetIndexFormat() == AZ::RHI::IndexFormat::Uint16 ? 2 : 4;
```

Comment:
This information needs to be exposed somehow, otherwise, if a (compute) shader accesses this with a ByteAddressBuffer, the offset-calculation will be wrong.

I suggest a function `ShaderStreamBufferViewsBuilder::GetIndexElementSize()` for now; actually dealing with this needs to happen elsewhere anyway

Reply:
ShaderStreamBufferViewsBuilder::BuildShaderIndexBufferView(...) is a private function, not exposed as a public API.
That's why the public API has the pairs:
For vertex indices: GetVertexIndicesBufferView() + GetVertexIndicesIndexBufferView().
For shader input streams: GetBufferView() + GetStreamBufferView()